pages/api/register.ts

```diff
@@ -1,22 +1,23 @@
-import { NextApiHandler } from "next";
+import type { Handler } from "retes";
 
-import { setAuthToken } from "../../lib/environment";
+import { Response } from 'retes/response';
+import { toNextHandler } from "retes/adapter";
+import { withMethod } from "retes/middleware";
 
-const handler: NextApiHandler = async (request, response) => {
-  const saleor_domain = request.headers["saleor-domain"];
-  if (!saleor_domain) {
-    response.status(400).json({ success: false, message: "Missing saleor domain token." });
-    return;
-  }
+import { setAuthToken } from "../../lib/environment";
+import { withAuthTokenRequired, withSaleorDomainPresent } from "@saleor/app-sdk/middleware";
 
-  const auth_token = request.body?.auth_token as string;
-  if (!auth_token) {
-    response.status(400).json({ success: false, message: "Missing auth token." });
-    return;
-  }
+const handler: Handler = async (request) => {
 
+  const auth_token = request.params.auth_token;
   await setAuthToken(auth_token);
-  response.json({ success: true });
+
+  return Response.OK({ success: true });
 };
 
-export default handler;
+export default toNextHandler([
+  withMethod('POST'),
+  withSaleorDomainPresent,
+  withAuthTokenRequired,
```

Comment:
we have `withSaleorDomainPresent`, so it would be more coherent to have `withAuthTokenPresent` instead of `withAuthTokenRequired`

Reply:
(but I also see that saleor domain is set as a header and auth token is part of a payload)

Reply:
(also we could start thinking about parametrizing that, like we have `withSaleorEventMatch("order_created")` in webhook handler)